### PR TITLE
Update testlab docs

### PR DESCRIPTION
**Pull request by loay**
connect to https://github.com/strongloop/loopback-next/issues/260
cc @bajtos @raymondfeng @ritch @superkhau

### CODE REVIEW

File: packages/testlab/README.md [hunk 1]
```diff
@@ -2,10 +2,19 @@
...
 
-## `expect`
+## Overview
 
-[Should.js](https://shouldjs.github.io/) configured in "as-function" mode
-(global `Object.prototype` is left intact) with an extra chaining word `to`.
+Test utilities to help writing loopback-next tests:
+- Should.js: expect(actual, expected, [message]).
+- sinon: Standalone test spies, stubs and mocks, that Works with any unit testing framework.
+
```
Comment (bajtos) [inline packages/testlab/README.md]: There are more utilities than just should + sinon: 

 - Helpers for creating `supertest` clients, most notably `createClientForApp` - see [client.ts](https://github.com/strongloop/loopback-next/blob/290fc7a60ac2ebe05826117f6788f99aab433d2f/packages/testlab/src/client.ts)
 - HTTP request/response mocks, see [shot.ts](https://github.com/strongloop/loopback-next/blob/290fc7a60ac2ebe05826117f6788f99aab433d2f/packages/testlab/src/shot.ts)

File: packages/testlab/README.md [hunk 2]
```diff
@@ -2,10 +2,19 @@
 
 A collection of test utilities we use to write LoopBack tests.
 
-## `expect`
+## Overview
 
-[Should.js](https://shouldjs.github.io/) configured in "as-function" mode
-(global `Object.prototype` is left intact) with an extra chaining word `to`.
+Test utilities to help writing loopback-next tests:
+- Should.js: expect(actual, expected, [message]).
```
Comment (bajtos) [inline packages/testlab/README.md]: I am proposing a slightly different description:

```md
- `expect` - BDD style assertions
```

File: packages/testlab/README.md [hunk 3]
```diff
@@ -2,10 +2,19 @@
...
+- sinon: Standalone test spies, stubs and mocks, that Works with any unit testing framework.
```
Comment (bajtos) [inline packages/testlab/README.md]: Let's shorten this to `sinon: test spies, stubs and mocks`

File: packages/testlab/README.md [hunk 4]
```diff
@@ -2,10 +2,19 @@
...
+
+## Installation
+
+```
+$ npm install --save @loopback/testlab
```
Comment (bajtos) [inline packages/testlab/README.md]: This is module is almost always used for tests only, I think we should be telling users to use `--save-dev`.

@crandmck any objections?

File: packages/testlab/README.md [hunk 5]
```diff
@@ -2,10 +2,19 @@
...
+```
+
+## Basic use
 
 #### Example usage
```
Comment (bajtos) [inline packages/testlab/README.md]: Please remove this header, `Basic use` is enough.

File: packages/testlab/README.md [hunk 6]
```diff
@@ -16,6 +25,41 @@ expect({key:'value'}).to.deepEqual({key: 'value'});
...
 
-## `sinon`
+```ts
+it('calls the original function', function () {
+    var callback = sinon.spy();
+    var proxy = once(callback);
+
+    proxy();
+
+    assert(callback.called);
```
Comment (bajtos) [inline packages/testlab/README.md]: 1) LoopBack apps should use `expect`, therefore our code examples should do the same.
2) I don't think it's necessary to show all helpers in the "Basic use" section. Let's stick to a single example that we had in "Example usage".

File: packages/testlab/README.md (hunk 4 above)
Comment (bajtos) [inline packages/testlab/README.md]: Could you please add an explanatory comment as you originally did in #480?

```markdown
_This package is typically used in tests, save it to `devDependencies` via `--save-dev`._
```

File: packages/testlab/README.md (hunk 2 above)
Comment (crandmck) [inline packages/testlab/README.md]: Please spell out "BDD" on first use.

File: packages/testlab/README.md (hunk 3 above)
Comment (crandmck) [inline packages/testlab/README.md]: I think we need some explanation of what "test spies" are... Stubs and mocks are perhaps better understood, but a short description would also help.  Also watch captialization (e.g. "...Works with...").

Oh, I see there's a link to http://sinonjs.org/ .  That may be sufficient, but I still think some more general info would help.

**Comment by loay** [comment]
@bajtos PTAL again. Thanks.

File: packages/testlab/README.md [hunk 7]
```diff
@@ -2,12 +2,26 @@
 
 A collection of test utilities we use to write LoopBack tests.
 
-## `expect`
+## Overview
 
-[Should.js](https://shouldjs.github.io/) configured in "as-function" mode
-(global `Object.prototype` is left intact) with an extra chaining word `to`.
+Test utilities to help writing loopback-next tests:
+- `expect` - behavior-driven development(BDD) style assertions
```
Comment (bajtos) [inline packages/testlab/README.md]: please add space before `(BDD)`

File: packages/testlab/README.md [hunk 8]
```diff
@@ -2,12 +2,26 @@
...
+- `sinon`
+ - stubs and mocks
+ - test spies: A function that records arguments, return value, the value of `this` and exception thrown for all its calls
```
Comment (bajtos) [inline packages/testlab/README.md]: While we are expanding `sinon` description, let's describe stubs and mocks for consistency.

```markdown
- test spies: functions recording arguments and other information for all of their calls
- stubs: functions (spies) with pre-programmed behavior
- mocks: fake methods (like spies) with pre-programmed behavior (like stubs) as well as pre-programmed expectations
```

File: packages/testlab/README.md [hunk 9]
```diff
@@ -2,12 +2,26 @@
...
+- Helpers for creating `supertest` clients, most notably createClientForApp
```
Comment (bajtos) [inline packages/testlab/README.md]:
```
- Helpers for creating `supertest` clients for LoopBack applications
```

File: packages/testlab/README.md [hunk 10]
```diff
@@ -2,12 +2,26 @@
...
+- HTTP request/response mocks
```
Comment (bajtos) [inline packages/testlab/README.md]:
```
- HTTP request/response stubs for writing tests without a listening HTTP server
```

**Comment by loay** [comment]
@bajtos PTAL again ..thanks
